python/guide/module/gluon/init.md

```diff
@@ -161,3 +170,127 @@ net.add(nn.Dense(10, in_units=256))
 
 net.initialize(init=MyInit())
 ```
+
+## Parameter Initialization
+
+Now that we know how to access the parameters, let's look at how to initialize
+them properly. By default, MXNet initializes the weight matrices
+uniformly by drawing random values with uniform-distribution between $-0.07$
+and $0.07$ ($U[-0.07, 0.07]$) and updates the bias parameters by setting them
+all to $0$.  However, we often need to use other methods to initialize the
+weights.  MXNet's `init` module provides a variety of preset initialization
+methods, but if we want something out of the ordinary, we need a bit of extra
+work.
+
+### Built-in Initialization
+
+Let's begin with the built-in initializers. The code below initializes all
+parameters with Gaussian random variables.
+
+```{.python .input  n=9}
+# force_reinit ensures that the variables are initialized again, regardless of whether they were
+# already initialized previously.
+net.initialize(init=init.Normal(sigma=0.01), force_reinit=True)
+net[0].weight.data()[0]
+```
+
+If we wanted to initialize all parameters to $1$, we could do this simply by
+changing the initializer to `Constant(1)`.
+
+```{.python .input  n=10}
+net.initialize(init=init.Constant(1), force_reinit=True)
+net[0].weight.data()[0]
+```
+
+If we want to initialize only a specific parameter in a different manner, we
+can simply set the initializer only for the appropriate subblock (or parameter)
+for that matter. For instance, below we initialize the second layer to a
+constant value of $42$ and we use the `Xavier` initializer for the weights of the
+first layer.
+
+```{.python .input  n=11}
+net[1].initialize(init=init.Constant(42), force_reinit=True)
+net[0].weight.initialize(init=init.Xavier(), force_reinit=True)
+print(net[1].weight.data()[0,0])
+print(net[0].weight.data()[0])
+```
+
+### Custom Initialization
+
+Sometimes, the initialization methods we need are not provided in the `init`
+module. At this point, we can implement a subclass of the `Initializer` class
+so that we can use it like any other initialization method. Usually, we only
+need to implement the `_init_weight` function and modify the incoming `NDArray`
```

Comment:
would reword: modify the incoming NDArray according to the initial result.

Reply:
This section was removed

Reply:
Simplified